packages/db/test/extensions/base.test.ts

```diff
@@ -0,0 +1,231 @@
+import { BlobStorage } from "@prisma/client";
+import { describe, it, expect } from "vitest";
+
+import { prisma } from "../../prisma";
+
+describe("Base Extension", () => {
+  describe("Address model", () => {
+    it("should upsert addresses from transactions", async () => {
```

Comment:
There are multiple cases we should test here:
- Check both fields are set correctly for addresses (sender and receiver) being stored for the first time
- Check an address already stored as a sender/receiver doesn't get its `firstBlockNumberAsSender`/`firstBlobNumberAsReceiver` field updated if the new block number is greater.